src/utils/trust/trust.involved.ts

```diff
@@ -0,0 +1,208 @@
+      beneficialOwnerTypeTitle: TRUST_INVOLVED_TEXTS.boTypeTitle,
+      trusteeTypeTitle: TRUST_INVOLVED_TEXTS.trusteeTypeTitle,
+      individualTrusteeData: getIndividualTrusteesFromTrust(appData, trustId),
+      formerTrusteeData: getFormerTrusteesFromTrust(appData, trustId),
+      trusteeType: TrusteeType,
+      checkYourAnswersUrl: getCheckYourAnswersUrl(isUpdate),
+      beneficialOwnerUrlDetach: `${config.TRUST_ENTRY_URL}/${trustId}${config.TRUST_BENEFICIAL_OWNER_DETACH_URL}`,
+    },
+    formData,
+    errors,
+    url: getServiceUrl(isUpdate),
+  };
+};
+
+const getTrustInvolvedPage = (
+  req: Request,
+  res: Response,
+  next: NextFunction,
+  isUpdate: boolean
+): void => {
+  try {
+    logger.debugRequest(req, `${req.method} ${req.route.path}`);
+
+    const pageProps = getPageProperties(req, isUpdate);
+
+    return res.render(pageProps.templateName, pageProps);
+  } catch (error) {
+    logger.errorRequest(req, error);
+    next(error);
+  }
+};
+
+const postTrustInvolvedPage = (
+  req: Request,
+  res: Response,
+  next: NextFunction,
+  isUpdate: boolean
+) => {
+  try {
+    logger.debugRequest(req, `${req.method} ${req.route.path}`);
+
+    if (req.body.noMoreToAdd) {
+      return safeRedirect(res, getPostTemplateName(isUpdate));
+    }
+    //  check on errors
+    const errorList = validationResult(req);
+
+    if (!errorList.isEmpty()) {
+      const pageProps = getPageProperties(
+        req,
+        isUpdate,
+        req.body,
+        formatValidationError(errorList.array()),
+      );
+
+      return res.render(pageProps.templateName, pageProps);
+    }
+
+    const typeOfTrustee = req.body.typeOfTrustee;
+
+    // the req.params['id'] is already validated in the has.trust.middleware but sonar can not recognise this.
+    let url = `${config.TRUST_ENTRY_URL}/${req.params[config.ROUTE_PARAM_TRUST_ID]}`;
+
+    switch (typeOfTrustee) {
+        case TrusteeType.HISTORICAL:
+          url += config.TRUST_HISTORICAL_BENEFICIAL_OWNER_URL;
+          break;
+        case TrusteeType.INDIVIDUAL:
+          url += config.TRUST_INDIVIDUAL_BENEFICIAL_OWNER_URL;
+          break;
+        case TrusteeType.LEGAL_ENTITY:
+          url += config.TRUST_LEGAL_ENTITY_BENEFICIAL_OWNER_URL;
+          break;
+        default:
+          logger.info("TODO: On validation No trustee selected, re-displaying page");
+    }
+
+    return safeRedirect(res, url);
+  } catch (error) {
+    logger.errorRequest(req, error);
+
+    return next(error);
+  }
+};
+
+export {
+  getTrustInvolvedPage as getTrustInvolvedPage,
+  postTrustInvolvedPage,
+  TRUST_INVOLVED_TEXTS,
+};
+
+const getTemplateName = (isUpdate: boolean) => {
+  if (isUpdate) {
+    return UPDATE_TRUSTS_INDIVIDUALS_OR_ENTITIES_INVOLVED_PAGE;
+  } else {
+    return TRUST_INVOLVED_PAGE;
+  }
+};
+
+const getBackLink = (isUpdate: boolean, trustId: string) => {
+  if (isUpdate) {
+    return UPDATE_TRUSTS_TELL_US_ABOUT_IT_URL;
```

Comment:
Do we need to add trustId so we know which trust?

Reply:
Yes I think we do, I'll add it now

Reply:
Fixed